lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/impl/NumberedSplit.java

```diff
@@ -62,6 +62,24 @@ public String toString() {
   }
 
   @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    NumberedSplit<?> that = (NumberedSplit<?>) o;
+    return index == that.index;
```

Comment:
Results of `equals()` must match results of `compareTo()`. In this case `compareTo()` is implemented based on index comparison only, so I can't use any other fields in `equals()`, or there will be mismatches.